awx/api/serializers.py

```diff
@@ -4690,33 +4706,13 @@ def get_summary_fields(self, obj):
                 if m2m_list:
                     summary_fields[fk] = []
                     for thisItem in m2m_list:
-                        if fk == 'job':
-                            summary_fields['job_template'] = []
-                            job_template_item = {}
-                            job_template_fields = SUMMARIZABLE_FK_FIELDS['job_template']
-                            job_template = getattr(thisItem, 'job_template', None)
-                            if job_template is not None:
-                                for field in job_template_fields:
-                                    fval = getattr(job_template, field, None)
-                                    if fval is not None:
-                                        job_template_item[field] = fval
-                                summary_fields['job_template'].append(job_template_item)
-                        if fk == 'workflow_job_template_node':
-                            summary_fields['workflow_job_template'] = []
-                            workflow_job_template_item = {}
-                            workflow_job_template_fields = SUMMARIZABLE_FK_FIELDS['workflow_job_template']
-                            workflow_job_template = getattr(thisItem, 'workflow_job_template', None)
-                            if workflow_job_template is not None:
-                                for field in workflow_job_template_fields:
-                                    fval = getattr(workflow_job_template, field, None)
-                                    if fval is not None:
-                                        workflow_job_template_item[field] = fval
-                                summary_fields['workflow_job_template'].append(workflow_job_template_item)
                         if fk == 'schedule':
                             unified_job_template = getattr(thisItem, 'unified_job_template', None)
                             if unified_job_template is not None:
                                 summary_fields[get_type_for_model(unified_job_template)] = {'id': unified_job_template.id,
                                                                                             'name': unified_job_template.name}
```

Comment:
This schedule case just makes so little sense. I really want this to be consolidated into the same helper.

What if instead of referencing `summary_keys[fk]` inside of `_job_method`, you passed in the job key. So the `if-else` here becomes `if-elif-elif` for the case of job, workflow job node, and then schedule. That would allow you to employ `get_type_for_model` during / before the call to `_job_method`.

Also, I realize that the schema is inconsistent for the case of schedules. We should prefer that to not be the case. Instead of doing id and name, we should let `_job_method` apply whatever fields it uses for the other cases (it is, in fact, summarizing the same object types). After doing this, it should still be confirmed that the UI links work after creating / modifying a schedule.